### PR TITLE
Azure Resource Naming Convention

### PR DESCRIPTION
**Pull request by charleszipp**
The document included provides a standardized naming convention for resources created within azure. This convention was used on my last engagement with a customer. It is also what we used at my previous employer (Wolters Kluwer S400 account). I believe this convention was provided to them by Microsoft at the start of their Azure journey. It worked well for us there as well as the previous customer engagement.

The benefits of this convention are

- Easily identify resource ownership across several different dimensions (module, location, env)
- Names are inherently unique to all of azure when using all 5 segments
- Ability to predict resource names without actually knowing the names. This is helpful when searching and running az cli commands
- Names are deterministic. This helps ensure deployments are idempotent when looking up resources by name.

The drawbacks are
- Usually gets pretty close to 24 character limit. Possible solution as suggested by Nick Drouin is to omit hyphens and use pascal or camel casing to save characters and maintain readability.

### CODE REVIEW

**Comment by DanMass** [comment]
Is there a link to a source for this standard or examples of it being used in the wild?

**Comment by DanMass** [comment]
The commit should include a link to the file from another navigable file in the playbook.

**Comment by charleszipp** [comment]
> The commit should include a link to the file from another navigable file in the playbook.

@DanMass For now I have included a link from the root readme. Unlike the docs in the playbook thus far, this one is tech specific. Therefor, I was having a difficult time trying to decide from which existing document it should be linked. Perhaps that's a sign the playbook isn't the right home for something like this? I am open to opinions/suggestions on hierarchy/structure for tech specific guidance within the playbook.

**Comment by charleszipp** [comment]
> Is there a link to a source for this standard or examples of it being used in the wild?

@DanMass 
This is the convention we used on a recent customer engagement led by Nick Drouin and Steven Goulet. I suggested this convention on this project as its what we used at my previous employer (S400 account). This was used in their division that was leading the rest of the company in Azure adoption. This convention was able to support over a hundred different modules across dozens of environments while still maintaining uniqueness and discoverability. I worked with this convention for over 2 years before coming to Microsoft. The pros and cons I observed during that time, as well as the customer engagement, have been outlined in the description of the PR.

Naming in general is difficult but, something that can be easily standardized in a best practice manner (i.e. its not a mandate but, documents what others have observed to be successful). I believe the biggest value here isn't the particular convention being suggested as much as it is having one in general. This would help teams move faster on engagements that are greenfield or involve new azure migrations by removing additional discussion/design over how to name azure resources. For those reasons, I am open to discussion on which particular convention to recommend.

File: Engineering/BestPractices/Azure/ResourceNamingConventions.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+| Resource Type             | Short Name |
+| ------------------------- | ---------- |
+| Resource Group            | rg         |
+| Storage Account           | st         |
+| Container Registry        | acr        |
+| Kubernetes Service        | aks        |
+| Key Vault                 | kv         |
+| Media Services            | mds        |
+| Media Services            | mds        |
```
Comment (polatengin) [inline Engineering/BestPractices/Azure/ResourceNamingConventions.md]: there is a duplication

File: Engineering/BestPractices/Azure/ResourceNamingConventions.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+| Application Insights      | ai         |
+| API Management Gateway    | apim       |
+| Service Fabric            | asf        |
+| Container Instances       | aci        |
+| SQL Server                | sql        |
+| SQL Database              | sqldb      |
+| Cosmos                    | cos        |
```
Comment (polatengin) [inline Engineering/BestPractices/Azure/ResourceNamingConventions.md]: I would prefer "cosdb"

File: Engineering/BestPractices/Azure/ResourceNamingConventions.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+A 2-4 character short name for the type of resource.
+
+- rg    = Resource Group
+- st    = Storage Account
+- acr   = Azure Container Registry
+- aks   = Azure Kubernetes Service
+
+### Module
+
+A short name for the module that is responsible for deploying the resource. For example the graph module will include `gph' in this segment.
```
Comment (polatengin) [inline Engineering/BestPractices/Azure/ResourceNamingConventions.md]: typo in `gph`, closing backtick is a single-quote

File: Engineering/BestPractices/Azure/ResourceNamingConventions.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+The location short name to which the resource is assigned. Resources deployed in East US would have eastus in this segment.
+
+- East US = eus
+- South Central US = scus
+- West US = wus
+
+#### Global Resources
+
+For resources not assigned to a location, the location will be omitted. Service Principals & Application Insights are examples of location agnostic resources in Azure.
+
```
Comment (polatengin) [inline Engineering/BestPractices/Azure/ResourceNamingConventions.md]: Also, link to the official documentation (https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/naming-and-tagging) in the References section would be better

**Comment by fnocera** [comment]
Please could you update and open a new PR for this given the new playbook structure? Thank you!

**Pull request closed without merging**